patch-package
--- a/node_modules/openapi-client-axios/client.js
+++ b/node_modules/openapi-client-axios/client.js
@@ -449,6 +449,10 @@ var OpenAPIClientAxios = /** @class */ (function () {
                 return Object.entries(methods).map(function (_a) {
                     var method = _a[0], operation = _a[1];
                     var op = __assign(__assign({}, (typeof operation === 'object' ? operation : {})), { path: path, method: method });
+                    operation.operationId = operation.operationId.replace(
+                        /[^0-9A-Za-z_$]+/g,
+                        "_"
+                    );
                     if (pathObject.parameters) {
                         op.parameters = __spreadArrays((op.parameters || []), pathObject.parameters);
                     }
