diff --git a/node_modules/openapi-client-axios-ts/client.js b/node_modules/openapi-client-axios-ts/client.js
index c415e10..79a38c2 100644
--- a/node_modules/openapi-client-axios-ts/client.js
+++ b/node_modules/openapi-client-axios-ts/client.js
@@ -502,6 +502,10 @@ var OpenAPIClientAxios = /** @class */ (function () {
                 return Object.entries(methods).map(function (_a) {
                     var method = _a[0], operation = _a[1];
                     var op = __assign(__assign({}, operation), { path: path, method: method });
+                    operation.operationId = operation.operationId.replace(
+                      /[^0-9A-Za-z_$]+/g,
+                      "_"
+                    );
                     if (pathObject.parameters) {
                         op.parameters = __spreadArrays((op.parameters || []), pathObject.parameters);
                     }
